core-bundle/src/EventListener/MakeRedirectResponseUrlAbsoluteListener.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Terminal42\ServiceAnnotationBundle\Annotation\ServiceTag;
+
+/**
+ * This makes sure that any redirect response in the Contao frontend scope is at least path-absolute (see #3065).
+ *
+ * @ServiceTag("kernel.event_listener")
+ */
+class MakeRedirectResponseUrlAbsoluteListener
+{
+    private $scopeMatcher;
+
+    public function __construct(ScopeMatcher $scopeMatcher)
+    {
+        $this->scopeMatcher = $scopeMatcher;
+    }
+
+    public function __invoke(ResponseEvent $event): void
+    {
+        $response = $event->getResponse();
+
+        if (!$this->scopeMatcher->isFrontendMasterRequest($event) || !$response instanceof RedirectResponse) {
+            return;
+        }
+
+        $url = $response->getTargetUrl();
+
+        if (preg_match('@^(http|/)@i', $url)) {
```

Comment:
This check could result in false negatives I think, like for `httpisgreat.html`.

```suggestion
        if ($url[0] === '/' || parse_url($url, PHP_URL_SCHEME) !== null) {
```

Reply:
True, changed in 7f7a238ee

Reply:
Can the url be empty?

Reply:
> Can the url be empty?

No, that would lead to

```
InvalidArgumentException:
Cannot redirect to an empty URL.
```

Reply:
We should proably add `'' === $url ||` for that case. Or do we want to convert the empty string to `/`?

Reply:
You cannot create a `RedirectResponse` object with an empty string (and you cannot call `setTargetUrl` with an empty string). It would lead to aforementioned exception.